daemon/core/services/quagga.py

```diff
@@ -194,52 +178,36 @@ def generateQuaggaBoot(cls, node, services):
         return 1
     fi
 
-    flags=""
+    QUAGGA_BIN_DIR=$(searchforprog 'vtysh' $QUAGGA_BIN_SEARCH)
+    if [ "z$QUAGGA_BIN_DIR" = "z" ]; then
+        echo "ERROR: Quagga's 'vtysh' program not found in search path:"
+        echo "  $QUAGGA_BIN_SEARCH"
+        return 1
+    fi
 
-    if [ "$1" != "zebra" ]; then
-        waitforvtyfiles zebra.vty
+    FLOCK_BIN_DIR=$(searchforprog 'flock' $QUAGGA_BIN_SEARCH)
+    if [ "z$FLOCK_BIN_DIR" = "z" ]; then
+        echo "ERROR: Couldn't find the 'flock' utility in search path:"
+        echo "  $QUAGGA_BIN_SEARCH"
+        return 1
     fi
 
+    flags=""
+
     if [ "$1" = "xpimd" ] && \\
         grep -E -q '^[[:space:]]*router[[:space:]]+pim6[[:space:]]*$' $QUAGGA_CONF; then
         flags="$flags -6"
     fi
 
     $QUAGGA_SBIN_DIR/$1 $flags -u $QUAGGA_USER -g $QUAGGA_GROUP -d
-}
-
-bootvtysh()
-{
-    QUAGGA_BIN_DIR=$(searchforprog $1 $QUAGGA_BIN_SEARCH)
-    if [ "z$QUAGGA_BIN_DIR" = "z" ]; then
-        echo "ERROR: Quagga's '$1' daemon not found in search path:"
-        echo "  $QUAGGA_SBIN_SEARCH"
-        return 1
-    fi
-
-    vtyfiles="zebra.vty"
-    for r in rip ripng ospf6 ospf bgp babel; do
-        if grep -q "^router \<${r}\>" $QUAGGA_CONF; then
-            vtyfiles="$vtyfiles ${r}d.vty"
-        fi
-    done
-
-    if grep -E -q '^[[:space:]]*router[[:space:]]+pim6?[[:space:]]*$' $QUAGGA_CONF; then
-        vtyfiles="$vtyfiles xpimd.vty"
-    fi
-
-    # wait for Quagga daemon vty files to appear before invoking vtysh
-    waitforvtyfiles $vtyfiles
-
-    $QUAGGA_BIN_DIR/vtysh -b
+    # (re-)run 'vtysh -b' to configure new daemon from integrated config file:
+    $FLOCK_BIN_DIR/flock $QUAGGA_STATE_DIR $QUAGGA_BIN_DIR/vtysh -b
```

Comment:
Is it possible that this command will wait forever?

From flock(1): 

>        By default, if the
>        lock cannot be immediately acquired, flock  waits  until  the  lock  is
>        available.

It seems like this is not a possibility. **vtysh** would need to hang, and then the script would get stuck anyway.